components/camel-nats/src/main/java/org/apache/camel/component/nats/NatsProducer.java

```diff
@@ -108,16 +108,53 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
         } else {
             LOG.debug("Publishing to topic: {}", config.getTopic());
 
-            final NatsMessage.Builder builder = NatsMessage.builder()
-                    .data(body)
-                    .subject(config.getTopic())
-                    .headers(this.buildHeaders(exchange));
-
-            if (ObjectHelper.isNotEmpty(config.getReplySubject())) {
-                final String replySubject = config.getReplySubject();
-                builder.replyTo(replySubject);
+            if (config.isJetstreamEnabled() && this.connection.getServerInfo().isJetStreamAvailable()) {
+                LOG.info("JetStream is available");
+
+                JetStreamManagement jsm;
+                JetStream js;
+                try {
+                    jsm = this.connection.jetStreamManagement();
+                    js = jsm.jetStream();
+                    if(js == null) {
+                        jsm.addStream(StreamConfiguration.builder()
+                                .name(config.getJetstreamName())
+                                .subjects(config.getTopic())
+                                .build());
+                        js = jsm.jetStream();
+                    }
+                } catch (IOException | JetStreamApiException e) {
+                    throw new RuntimeException(e);
+                }
+                PublishAck pa;
+                if (config.isJetstreamAsync()) {
+                    CompletableFuture<PublishAck> future = js.publishAsync(config.getTopic(), body);
+                    try {
+                        pa = future.get(1, TimeUnit.SECONDS);
+                    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                        throw new RuntimeException(e);
+                    }
+                    LOG.info("Publish Sequence async: {}", pa.getSeqno());
+                } else {
+                    try {
+                        pa = js.publish(config.getTopic(), body);
+                    } catch (IOException | JetStreamApiException e) {
+                        throw new RuntimeException(e);
+                    }
+                    LOG.info("Publish Sequence synchronously: {}", pa.getSeqno());
+                }
+            } else {
```

Comment:
I would do the other way around. Jet stream will be not enabled by default, so its part should be in the else.

Reply:
Yes, it makes sense when changing default.